adjacent_motif_feature_generator.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import itertools
+import logging as log
+
+from common import mutate_string, NUCLEOTIDES
+from generic_feature_generator import GenericFeatureGenerator
+
+class AdjacentMotifFeatureGenerator(GenericFeatureGenerator):
+    """
+    Distances calculated from rightmost nucleotide of query motifs
+    """
+    def __init__(
+            self,
+            motif_len=4,
+            distance_to_right_flank_end=0,
+            feats_to_remove=[],
+            left_update_region=1,
+            right_update_region=1
+        ):
+        """
+        @param motif_len: list of motifs to look for at a distance from mutating position
+        @param distance_to_right_flank_end: distance from mutating position to look for nearby motif
+        @param feats_to_remove: motifs to remove
+        @param left_update_region: number of positions to consider left of mutating position to update
+            features correctly
+        @param right_update_region: number of positions to consider right of mutating position to update
+            features correctly
+        """
+        self.motif_len = motif_len
+        self.distance_to_right_flank_end = distance_to_right_flank_end
+        self.left_update_region = left_update_region
+        self.right_update_region = right_update_region
+
+        self.update_feats_after_removing(feats_to_remove)
+
+    def update_feats_after_removing(self, feats_to_remove=[]):
+        """
+        take existing SubmotifGenerator and update it with features to remove
+        """
+        all_feature_info_list = [("".join(motif), self.distance_to_right_flank_end) for motif in itertools.product(*([NUCLEOTIDES] * self.motif_len))]
+        self.feature_info_list = [feat_tuple for feat_tuple in all_feature_info_list if feat_tuple not in feats_to_remove]
+
+        self.motif_list = [motif for motif, _ in self.feature_info_list]
+        self.motif_dict = {motif: i for i, motif in enumerate(self.motif_list)}
+        for motif, dist in all_feature_info_list:
+            if (motif, dist) in feats_to_remove:
+                self.motif_dict[motif] = None
+
+        self.feature_vec_len = len(self.feature_info_list)
+
+    def print_label_from_info(self, info):
```

Comment:
maybe you can describe that this function prints things from `feature_info_list`?